### DISCUSSION TITLE
Test Coverage

### DISCUSSION THREAD

**Discussion by armanbilge** [http4s/http4s#6453] [General]
We briefly discussed enabling coverage in https://github.com/http4s/http4s/pull/6371#issuecomment-1146963559, but it seems that tracking coverage metrics per PR has some serious caveats.

On the other hand, it seems reasonable to me that we can run coverage now and again and see how we're doing overall, without linking the results to a specific change. So that's what I did.

View the report at either:
- https://armanbilge.github.io/http4s/scoverage-report/
- https://app.codecov.io/gh/armanbilge/http4s/

**Comment by rossabaker** [comment]
Any estimate how much it slows down builds?

I wonder whether this would be a good candidate for a weekly workflow.  Keep it fresh, but don't let it be abused.

**Reply by armanbilge** [reply]
Negligibly, the slowest part of our CI is currently the JS jobs (sorry :).

I think the best way is to run coverage as a side-job for a single Scala and JVM. Most code is shared, so this gives reasonable results. I've learned the hard way that artifacts compiled with coverage on should not be passed to the publish step 😅 

A weekly workflow could work as well 👍

**Reply by rossabaker** [reply]
Yeah, keeping coverage quarantined in a separate job is good.  I once lost a couple days to SBT plugin interactions turning coverage back on at surprising places back in The Rig Era.

**Reply by armanbilge** [reply]
This landed in https://github.com/http4s/http4s/pull/6456.
